src/Util/FunctionUtil.ts

```diff
@@ -0,0 +1,254 @@
+    if (func.name === 'property') {
+      if (!feature) {
+        throw new Error(`Could not evalute 'property' function. Feature ${feature} is not defined.`);
+      }
+      if (isGeoStylerStringFunction(func.args[0]) && feature?.properties) {
+        const arg = FunctionUtil.evaluateStringFunction(func.args[0], feature);
+        return feature?.properties?.[arg];
+      } else {
+        const arg = func.args[0] as string;
+        return feature?.properties?.[arg];
+      }
+    }
+
+    if (isGeoStylerStringFunction(func)) {
+      return FunctionUtil.evaluateStringFunction(func, feature) as ReturnType<T>;
+    }
+    if (isGeoStylerNumberFunction(func)) {
+      return FunctionUtil.evaluateNumberFunction(func, feature) as ReturnType<T>;
+    }
+    if (isGeoStylerBooleanFunction(func)) {
+      return FunctionUtil.evaluateBooleanFunction(func, feature) as ReturnType<T>;
+    }
+    if (isGeoStylerUnknownFunction(func)) {
+      return FunctionUtil.evaluateUnknownFunction(func, feature) as ReturnType<T>;
+    }
+    return undefined;
+  }
+
+  public static evaluateBooleanFunction(func: GeoStylerBooleanFunction, feature?: Feature): boolean {
+    const args = func.args.map(arg => {
+      if (isGeoStylerFunction(arg)) {
+        return FunctionUtil.evaluateFunction(arg, feature);
+      }
+      return arg;
+    });
+    switch (func.name) {
+      case 'between':
+        return (args[0] as number) >= (args[1] as number) && (args[0] as number) <= (args[2] as number);
+      case 'double2bool':
+        // TODO: evaluate this correctly
+        return false;
+      case 'in':
+        return args.slice(1).includes(args[0]);
+      case 'parseBoolean':
+        return !!args[0];
+      case 'strEndsWith':
+        return (args[0] as string).endsWith(args[1] as string);
+      case 'strEqualsIgnoreCase':
+        return (args[0] as string).toLowerCase() === (args[1] as string).toLowerCase() ;
+      case 'strMatches':
+        const regEx = (args[1] as string);
+        const regexArray = regEx.match(/\/(.*?)\/([gimy]{0,4})$/);
+        if (regexArray && regexArray.length === 3){
+          return new RegExp(regexArray[1], regexArray[2]).test(args[0] as string);
+        } else {
+          return false;
+        }
+      case 'strStartsWith':
+        return (args[0] as string).startsWith(args[1] as string);
+      default:
+        return false;
+    }
+  }
+
+  public static evaluateNumberFunction(func: GeoStylerNumberFunction, feature?: Feature): number {
+    if (func.name === 'pi') {
+      return Math.PI;
+    }
+    if (func.name === 'random') {
+      return Math.random();
+    }
+    const args = func.args.map(arg => {
+      if (isGeoStylerFunction(arg)) {
+        return FunctionUtil.evaluateFunction(arg, feature);
+      }
+      return arg;
+    });
+    switch (func.name) {
+      case 'abs':
+        return Math.abs(args[0] as number);
```

Comment:
What happens if no number is passed in the arguments? Should the args be checked here, e.g. using `_isFinite(…)`?

Reply:
This should not be a problem. This should just return `NaN` when you pass in an illegal `GeoStylerNumberFunction`. Classy shit in shit out. :smile: